ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java

```diff
@@ -236,4 +243,29 @@ static Options createOptions() {
 
     return result;
   }
+
+  /**
+   * While deleting dangling scratch dirs from hdfs, we can clean corresponding local files as well
+   * @param sessionName prefix to determine removable tmp files
+   * @param localTmpdir local tmp file location
+   */
+  public void removeLocalTmpFiles(String sessionName, String localTmpdir) {
+    File[] files = new File(localTmpdir).listFiles();
+    boolean success;
+    if (files != null) {
+      for (File file : files) {
+        success = false;
+        if (file.getName().startsWith(sessionName)) {
```

Comment:
You may use File.listFiles() method with a filter instead of this construct. It's not necessary to iterate through all the files again and again inside the outer loop.

Reply:
I mean the following:
`    File[] files = new File(localTmpdir).listFiles((fn -> fn.getName().startsWith(sessionName)));`
Using this approach you loop through only the files which needs to be deleted, so there is no need for filename check.

Reply:
This is done now. I have added `startsWith` check as provided and removed it from the loop. As you said now we do not deal with all the files in the loop.